test/spanner.ts

```diff
@@ -5441,6 +5655,81 @@ describe('Spanner with mock server', () => {
       });
     });
   });
+
+  describe('XGoogRequestId', () => {
+    it('with retry on aborted query', async () => {
+      let attempts = 0;
+      const database = newTestDatabase();
+      let rowCount = 0;
+      const maxAttempts = 4;
+      await database.runTransactionAsync(async transaction => {
+        attempts++;
+        if (attempts < maxAttempts) {
+          spannerMock.abortTransaction(transaction!);
+        }
+        const [rows] = await transaction!.run(selectSql);
+        rows.forEach(() => rowCount++);
+        assert.strictEqual(rowCount, 3);
+        assert.strictEqual(attempts, 4);
+        await transaction!.commit();
+      });
+
+      // TODO: introduce tests for incremented attempts to verify
```

Comment:
This test checks the behavior when a transaction is retried due to Spanner aborting a read/write transaction. That should not lead to the attempt number increasing, meaning that there is no pending TODO in this test. The current behavior should remain exactly the same as what is verified in this test now, also when support for increasing the attempt number has been implemented.

The attempt number should only be bumped when a single RPC is retried by Gax.

Reply:
Oh yes, I just put the TODO inside the test I had included blindly. Let me make an update then ping. Thanks for spotting that.